src/ctia/stores/atom/common.clj

```diff
@@ -7,59 +7,50 @@
   (fn [_new-entity_]
     (str prefix "-" (UUID/randomUUID))))
 
-(defmacro def-read-handler [name Model]
-  `(s/defn ~name :- (s/maybe ~Model)
-     [state# :- (s/atom {s/Str ~Model})
-      id# :- s/Str]
-     (get (deref state#) id#)))
+(defn read-handler [Model]
+  (s/fn :- (s/maybe Model)
```

Comment:
You could also not use schema/fn, and just use clojure.core/fn if concerned about the performance hit in this one spot. Though how much does performance actually matter here?